src/types.jl

```diff
@@ -149,12 +507,552 @@ for Model in [:NeuralNetworkRegressor, :MultitargetNeuralNetworkRegressor]
+  scitype is `Continuous`; check the scitype with `schema(y)`
+
+
+# Hyper-parameters
+
+- `builder=MLJFlux.Linear(σ=Flux.relu)`: An MLJFlux builder that constructs a neural
+  network. Possible `builders` include: `Linear`, `Short`, and `MLP`. You can construct
+  your own builder using the `@builder` macro, see examples for further information.
+- `optimiser::Flux.ADAM()`: A `Flux.Optimise` optimiser. The optimiser performs the
+  updating of the weights of the network. For further reference, see either the examples
+  or [the Flux optimiser
+  documentation](https://fluxml.ai/Flux.jl/stable/training/optimisers/). To choose a
+  learning rate (the update rate of the optimizer), a good rule of thumb is to start out
+  at `10e-3`, and tune using powers of 10 between `1` and `1e-7`.
+- `loss=Flux.mse`: The loss function which the network will optimize. Should be a
+  function which can be called in the form `loss(yhat, y)`. Possible loss functions are
+  listed in [the Flux loss function
+  documentation](https://fluxml.ai/Flux.jl/stable/models/losses/). For a regression task,
+  the most natural loss functions are:
+    - `Flux.mse`
+    - `Flux.mae`
+    - `Flux.msle`
+    - `Flux.huber_loss`
+- `epochs::Int=10`: The number of epochs to train for. Typically, one epoch represents
+  one pass through the entirety of the training dataset.
+- `batch_size::Int=1`: The batch size to be used for training. The batch size represents
+  the number of samples per update of the networks weights. Typcally, batch size should be
+  somewhere between 8 and 512. Smaller batch sizes lead to noisier training loss curves,
+  while larger batch sizes lead towards smoother training loss curves. In general, it is a
+  good idea to pick one fairly large batch size (e.g. 32, 64, 128), and stick with it, and
+  only tune the learning rate. In most literature, batch size is set in powers of twos,
+  but this is fairly arbitrary.
+- `lambda::Float64=0`: The stregth of the regularization used during training. Can be
+  any value  in the range `[0, ∞)`.
+- `alpha::Float64=0`: The L2/L1 mix of regularization, in the range `[0, 1]`. A value of
+  0 represents L2 regularization, and a value of 1 represents L1 regularization.
+- `rng::Union{AbstractRNG, Int64}`: The random number generator/seed used during
+  training.
+- `optimizer_changes_trigger_retraining::Bool=false`: Defines what happens when fitting
+  a machine if the associated optimiser has changed. If true, the associated machine will
+  retrain from scratch on `fit`, otherwise it will not.
+- `acceleration::AbstractResource=CPU1()`: Defines on what hardware training is done.
+  For Training on GPU, use `CudaLibs()`, otherwise defaults to `CPU`()`.
+- `finaliser=Flux.softmax`: The final activation function of the neural network.
+Defaults to `Flux.softmax`. For a regression task, reasonable alternatives include
+`Flux.sigmoid` and the identity function (otherwise known as "linear activation").
+
```

Comment:
Please revise the `batch_size` and `loss` fields as noted above for nn classifier